### PR TITLE
[DOC] Added dalapi tool

### PR DESCRIPTION
**Pull request by outoftardis**
# Description

- Transferred dalapi extension from spec: only the tool and configs for it from #1074. Other changes will be added through other PRs.
- Updated CI for docs to incorporate doxygen into the build process

### CODE REVIEW

**Comment by rlnx** [comment]
We need to update license in dalapi files to Apache

File: docs/Makefile [hunk 1]
```diff
@@ -31,9 +31,15 @@ create_rst_examples:
...
+
+parse-doxygen: doxygen
+	mkdir -p build
+
+.PHONY: help Makefile create_rst_examples doxygen parse-doxygen
 
 # Catch-all target: route all unknown targets to Sphinx using the new
 # "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
-%: create_rst_examples Makefile
+%: create_rst_examples doxygen parse-doxygen Makefile
```
Comment (rlnx) [inline docs/Makefile:44]: Remove dependency on doxygen and parse-doxygen

File: docs/.gitignore [hunk 2]
```diff
@@ -2,3 +2,6 @@ build/
 source/examples/cpp/*
 source/examples/dpc/*
 docbuild-log.txt
+__pycache__
+doxygen/oneapi/doxygen/xml
+.vs*
```
Comment (rlnx) [inline docs/.gitignore:7]: Isn't .vs defined by gitignore in the repo root, if it's not, I'd recommend to put it there

File: docs/Makefile (hunk 1 above)
Comment (outoftardis) [reply docs/Makefile:45]: done

File: docs/Makefile [hunk 3]
```diff
@@ -31,9 +31,15 @@ create_rst_examples:
 	@echo "Creating rST files for oneAPI Examples..."
 	python3 rst_examples.py
 
-.PHONY: help Makefile create_rst_examples
+doxygen:
+	doxygen doxygen/oneapi/Doxyfile
+
+parse-doxygen: doxygen
+	mkdir -p build
```
[the remark this answers is not in the document]
Comment (outoftardis) [reply docs/Makefile:38]: done

File: docs/.gitignore (hunk 2 above)
Comment (outoftardis) [reply docs/.gitignore]: done

**Comment by outoftardis** [comment]
@napetrov I think some exceptions for our CI should be added on the infra side

**Comment by outoftardis** [comment]
@rlnx @napetrov what what I see, CI fails due to unrelated issues. Can we merge?

**Comment by napetrov** [comment]
yes, we can merge

**Review by napetrov — APPROVED** [review]

**Pull request merged**